crd-generator/api/src/test/java/io/fabric8/crd/generator/v1/JsonSchemaTest.java

```diff
@@ -178,15 +179,58 @@ void shouldExtractPropertiesSchemaFromExtractValueAnnotation() {
     assertNull(barProps.get("baz"));
   }
 
+  @Test
+  void shouldExtractPropertiesSchemaFromSchemaSwapAnnotations() {
+    TypeDef extraction = Types.typeDefFrom(MultipleSchemaSwaps.class);
+    JSONSchemaProps schema = JsonSchema.from(extraction);
+    assertNotNull(schema);
+    Map<String, JSONSchemaProps> properties = schema.getProperties();
+    assertEquals(2, properties.size());
+    final JSONSchemaProps specSchema = properties.get("spec");
+    Map<String, JSONSchemaProps> spec = specSchema.getProperties();
+    assertEquals(3, spec.size());
+
+    // 'first' is replaced by SchemaSwap from int to string
+    JSONSchemaProps first = spec.get("first");
+    Map<String, JSONSchemaProps> firstProps = first.getProperties();
+    assertNotNull(firstProps);
+    JSONSchemaProps firstProperty = firstProps.get("shouldBeString");
+    assertEquals("string", firstProperty.getType());
+
+    // 'second' is replaced by the same SchemaSwap that is applied multiple times
+    JSONSchemaProps second = spec.get("second");
+    Map<String, JSONSchemaProps> secondProps = second.getProperties();
+    assertNotNull(secondProps);
+    JSONSchemaProps secondProperty = secondProps.get("shouldBeString");
+    assertEquals("string", secondProperty.getType());
+
+    // 'third' is replaced by the another SchemaSwap
+    JSONSchemaProps third = spec.get("third");
+    Map<String, JSONSchemaProps> thirdProps = third.getProperties();
+    assertNotNull(thirdProps);
+    JSONSchemaProps thirdProperty = thirdProps.get("shouldBeInt");
+    assertEquals("integer", thirdProperty.getType());
+  }
+
   @Test
   void shouldThrowIfSchemaSwapHasUnmatchedField() {
     TypeDef incorrectExtraction = Types.typeDefFrom(IncorrectExtraction.class);
-    assertThrows(IllegalArgumentException.class, () -> JsonSchema.from(incorrectExtraction));
+    IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+        () -> JsonSchema.from(incorrectExtraction));
+    assertEquals(
+        "Unmatched SchemaSwaps: @SchemaSwap(originalType=io.fabric8.crd.example.extraction.ExtractionSpec, fieldName=\"FOO\", targetType=io"
+            + ".fabric8.crd.example.extraction.FooExtractor) on io.fabric8.crd.example.extraction.IncorrectExtraction",
+        exception.getMessage());
```

Comment:
The new implementation no longer differentiates between unmatched class and unmatched field. I my opinion the more detailed message compensates for it.

Reply:
totally fine to change the message, but I believe that this change of behavior is caused by the fact that we are missing an invariant, see the comment above on matching the field.